### PR TITLE
Challenges: classmethods added to Challenge and ChallengePhase Model

### PR DESCRIPTION
**Pull request by deshraj**

### CODE REVIEW

File: apps/challenges/models.py [hunk 1]
```diff
@@ -54,3 +87,22 @@ class ChallengePhase(TimeStampedModel):
...
+    def get_end_date(self):
+        """Returns the end date of Challenge"""
+        return self.end_date
+
+    @property
+    def is_active(self):
+        """Returns if the challenge is active or not"""
+        if self.end_date > timezone.now():
+            return True
+        return False
```
Comment (dexter1691) [inline apps/challenges/models.py]: are we planning to put get methods for each field? 
If yes, then he getter method list seems incomplete. 
If not, then we should probably get rid of all get functions, right?

Comment (dexter1691) [inline apps/challenges/models.py]: @deshraj: Can we add a check that end_date should always be greater than start_date? 
Also write a corresponding test case for that!

File: tests/unit/challenges/test_models.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    def setUp(self):
+        super(ChalengeTestCase, self).setUp()
+
+        try:
+            os.makedirs('/tmp/evalai')
+        except OSError:
+            pass
+
+        with self.settings(MEDIA_ROOT='/tmp/evalai'):
+            self.challenge.image = SimpleUploadedFile('test_sample_file.jpg',
```
Comment (dexter1691) [inline tests/unit/challenges/test_models.py]: Probably more rigorous tests here would be nice! For example: 
Can we check if random characters return an error or not? 
Can we check if the length of the URL is okay? 

We should probably put a check for folder name being not too long! and image name being not too long!

Comment (dexter1691) [inline tests/unit/challenges/test_models.py]: ^Those are probably for views.py though. So feel free to ignore these for models.py and make sure we include them in other tests.

File: apps/challenges/models.py (hunk 1 above)
Comment (deshraj) [inline apps/challenges/models.py]: No, we are not planning to have get method for each field. We will have methods for those fields which we need to use somewhere and some other methods that will devise some metric using the fields already present such as `is_active()`. 

Sure, I will add a test for that.

**Pull request merged**